File: luaInvoking/FileSystem.cpp
#include "FileSystem.h"
using namespace XXHelper;

const char FileSystem::className[] = "FileSystem22";
Lunar<FileSystem>::RegType FileSystem::methods[] =
{
	LUNAR_DECLARE_METHOD(FileSystem, MoveFile),
	LUNAR_DECLARE_METHOD(FileSystem, GetCurrentDirectory),
	LUNAR_DECLARE_METHOD(FileSystem, CopyFile),
	LUNAR_DECLARE_METHOD(FileSystem, GetFileInfo),
	LUNAR_DECLARE_METHOD(FileSystem, BuildPath),
	LUNAR_DECLARE_METHOD(FileSystem, GetExtension),
	LUNAR_DECLARE_METHOD(FileSystem, Foreach),
	LUNAR_DECLARE_METHOD(FileSystem, GetSysPath),
	LUNAR_DECLARE_METHOD(FileSystem, CreateFolder),
	LUNAR_DECLARE_METHOD(FileSystem, DeleteFile),
	LUNAR_DECLARE_METHOD(FileSystem, RenameFile),
	LUNAR_DECLARE_METHOD(FileSystem, DeleteFolder),
	LUNAR_DECLARE_METHOD(FileSystem, GetDirectoryPath),
	LUNAR_DECLARE_METHOD(FileSystem, SetCurrentDirectory),
	LUNAR_DECLARE_METHOD(FileSystem, FileExist),
	LUNAR_DECLARE_METHOD(FileSystem, GetFileName),
	LUNAR_DECLARE_METHOD(FileSystem, SearchFile),
	LUNAR_DECLARE_METHOD(FileSystem, GetPathInfo),
	LUNAR_DECLARE_METHOD(FileSystem, StrReplacePathSpliteToUnixStyle),
	LUNAR_DECLARE_METHOD(FileSystem, GetFileNameWithoutExtension),
	LUNAR_DECLARE_METHOD(FileSystem, PackBundle),
	LUNAR_DECLARE_METHOD(FileSystem, UnpackBundle),
	LUNAR_DECLARE_METHOD(FileSystem, SearchFolder),
	{0,0}
};

FileSystem::FileSystem(void){}
FileSystem::FileSystem(lua_State* L){}
FileSystem::~FileSystem(void){}

int FileSystem::MoveFile(lua_State* lua)
{
	printf("FileSystem::MoveFile:\n");
	return 0;
}
int FileSystem::GetCurrentDirectory(lua_State* lua)
{
	return 0;
}
int FileSystem::CopyFile(lua_State* lua)
{
	return 0;
}
int FileSystem::GetFileInfo(lua_State* lua)
{
	return 0;
}
int FileSystem::BuildPath(lua_State* lua)
{
	return 0;
}
int FileSystem::GetExtension(lua_State* lua)
{
	return 0;
}
int FileSystem::Foreach(lua_State* lua)
{
	return 0;
}
int FileSystem::GetSysPath(lua_State* lua)
{
	return 0;
}
int FileSystem::CreateFolder(lua_State* lua)
{
	return 0;
}
int FileSystem::DeleteFile(lua_State* lua)
{
	return 0;
}
int FileSystem::RenameFile(lua_State* lua)
{
	return 0;
}
int FileSystem::DeleteFolder(lua_State* lua)
{
	return 0;
}
int FileSystem::GetDirectoryPath(lua_State* lua)
{
	return 0;
}
int FileSystem::SetCurrentDirectory(lua_State* lua)
{
	return 0;
}
int FileSystem::FileExist(lua_State* lua)
{
	return 0;
}
int FileSystem::GetFileName(lua_State* lua)
{
	return 0;
}
int FileSystem::SearchFile(lua_State* lua)
{
	return 0;
}
int FileSystem::GetPathInfo(lua_State* lua)
{
	return 0;
}
int FileSystem::StrReplacePathSpliteToUnixStyle(lua_State* lua)
{
	return 0;
}
int FileSystem::GetFileNameWithoutExtension(lua_State* lua)
{
	return 0;
}
int FileSystem::PackBundle(lua_State* lua)
{
	return 0;
}
int FileSystem::UnpackBundle(lua_State* lua)
{
	return 0;
}
int FileSystem::SearchFolder(lua_State* lua)
{
	return 0;
}
bool FileSystem::_MoveFile(string src,string dst,bool bOverride)
{
	return true;
}
bool FileSystem::_SetCurrentDirectory(string path)
{
	return true;
}
string FileSystem::_GetCurrentDirectory()
{
	return "";
}
bool FileSystem::_CopyFile(string src,string dst,bool bOverride)
{
	return true;
}
bool FileSystem::_GetFileInfo(string filePath)
{
	return true;
}
bool FileSystem::_BuildPath(string path)
{
	return true;
}
string FileSystem::_GetExtension(string filePath)
{
	return "";
}
bool FileSystem::_Foreach()
{
	return true;
}
bool FileSystem::_GetSysPath()
{
	return true;
}
bool FileSystem::_CreateFolder()
{
	return true;
}
bool FileSystem::_DeleteFile(string filePath)
{
	return true;
}
bool FileSystem::_RenameFile()
{
	return true;
}
bool FileSystem::_DeleteFolder(string path,bool bRecursive)
{
	return true;
}
string FileSystem::_GetDirectoryPath(string filePath)
{
	return "";
}
bool FileSystem::_FileExist()
{
	return true;
}
string FileSystem::_GetFileName(string filePath)
{
	return "";
}
bool FileSystem::_SearchFile(string pattern,bool bFullPath,bool bSearchHide)
{
	return true;
}
bool FileSystem::_GetPathInfo(string path)
{
	return true;
}
bool FileSystem::_StrReplacePathSpliteToUnixStyle()
{
	return true;
}
string FileSystem::_GetFileNameWithoutExtension(string filePath)
{
	return "";
}
bool FileSystem::_PackBundle()
{
	return true;
}
bool FileSystem::_UnpackBundle()
{
	return true;
}
bool FileSystem::_SearchFolder(string path,bool bRecursive,bool bSearchHide)
{
	return true;
}

File: Project/FileSystem.lua

_G.FileSystem={}

-- 设置当前的相对路径
function FileSystem:SetCurrentDirectory(path)

end

-- 得到当前路径
function FileSystem:GetCurrentDirectory()

end

-- 搜索文件
-- eg local files = FileSystem:SearchFile("D:/works/*.log");
-- 
function FileSystem:SearchFile(pattern,bFullPath,bSearchHide)

end

-- 搜索指定目录下的子目录
function FileSystem:SearchFolder(path,bRecursive,bSearchHide)

end

-- 得到一个文件所在的文件夹
-- eg: local folder = FileSystem:GetDirectoryPath("D:/works/hello.log");
-- "folder=D:/works"
function FileSystem:GetDirectoryPath(filePath)

end

-- 得到文件名
-- eg: print(FileSystem:GetFileName("D:/works/hello.log")); 返回 hello.log
function FileSystem:GetFileName(filePath)

end

-- 得到不带扩展名的文件名
function FileSystem:GetFileNameWithoutExtension(filePath)

end

-- 得到扩展名部分
function FileSystem:GetExtension(filePath)

end

-- 得到文件信息,如文件大小，创建日期,最后一次访问时间，以及修改时间,是否隐藏，文件类型等
-- return fileInfo,code,errInfo
function FileSystem:GetFileInfo(filePath)

end

-- 返回路径对应的文件信息
-- directory 如果path是文件夹
-- file 如果path是文件
-- nil 如果路径不对应任何文件
-- invalid 如果是无效的路径
function FileSystem:GetPathInfo(path)

end

-- 移动文件,返回0成功没有错误
-- 返回code,errInfo
function FileSystem:MoveFile(src,dst,bOverride)

end

-- 复制文件到指定目录
-- return code,errInfo
function FileSystem:CopyFile(src,dst,bOverride)

end

-- 删除文件
-- return code,errInfo
function FileSystem:DeleteFile(filePath)

end

-- 删除文件夹
-- path:要删除的路径必须是个文件夹
-- bRecursive:是否递归删除子目录
-- return: code, errInfo
function FileSystem:DeleteFolder(path,bRecursive)

end

-- 更具指定的路径构建文件夹目录
function FileSystem:BuildPath(path)

end

-- 创建文件夹
-- return code,errInfo
function FileSystem:CreateFolder(folderPath)

end

-- 将文件打包到一个指定地址
-- return code, errInfo
function FileSystem:PackPackage(folder,dstPath)

end

-- 将压缩包解压到指定文件位置
function FileSystem:UnpackPackage(packagePath,dstPath)
	
end
